pkg/agent/plugin/workloadattestor/k8s/k8s.go

```diff
@@ -152,42 +153,272 @@ func (p *k8sPlugin) Attest(ctx context.Context, req *workloadattestor.AttestRequ
 
 		// if the container was not located and there were no pods with
 		// uninitialized containers, then the search is over.
-		if !notAllContainersReady || attempt >= p.maxPollAttempts {
-			log.Printf("container id %q not found (attempt %d of %d)", containerID, attempt, p.maxPollAttempts)
+		if !notAllContainersReady || attempt >= config.MaxPollAttempts {
+			log.Printf("container id %q not found (attempt %d of %d)", containerID, attempt, config.MaxPollAttempts)
 			return nil, k8sErr.New("no selectors found")
 		}
 
 		// wait a bit for containers to initialize before trying again.
-		log.Printf("container id %q not found (attempt %d of %d); trying again in %s", containerID, attempt, p.maxPollAttempts, p.pollRetryInterval)
+		log.Printf("container id %q not found (attempt %d of %d); trying again in %s", containerID, attempt, config.MaxPollAttempts, config.PollRetryInterval)
 
 		select {
-		case <-time.After(p.pollRetryInterval):
+		case <-p.clock.After(config.PollRetryInterval):
 		case <-ctx.Done():
 			return nil, k8sErr.New("no selectors found: %v", ctx.Err())
 		}
 	}
 }
 
-func (p *k8sPlugin) getPodListFromInsecureKubeletPort() (out *podList, err error) {
-	httpResp, err := p.httpClient.Get(fmt.Sprintf("http://localhost:%d/pods", p.kubeletReadOnlyPort))
+func (p *k8sPlugin) Configure(ctx context.Context, req *spi.ConfigureRequest) (resp *spi.ConfigureResponse, err error) {
+	// Parse HCL config payload into config struct
+	config := new(k8sHCLConfig)
+	if err := hcl.Decode(config, req.Configuration); err != nil {
+		return nil, k8sErr.New("unable to decode configuration: %v", err)
+	}
+
+	// Determine max poll attempts with default
+	maxPollAttempts := config.MaxPollAttempts
+	if maxPollAttempts <= 0 {
+		maxPollAttempts = defaultMaxPollAttempts
+	}
+
+	// Determine poll retry interval with default
+	var pollRetryInterval time.Duration
+	if config.PollRetryInterval != "" {
+		pollRetryInterval, err = time.ParseDuration(config.PollRetryInterval)
+		if err != nil {
+			return nil, k8sErr.New("unable to parse poll retry interval: %v", err)
+		}
+	}
+	if pollRetryInterval <= 0 {
+		pollRetryInterval = defaultPollRetryInterval
+	}
+
+	// Determine which kubelet port to hit. Default to the secure port if none
+	// is specified (this is backwards compatible because the read-only-port
+	// config value has always been required, so it should already be set in
+	// existing configurations that rely on it).
+	if config.KubeletSecurePort > 0 && config.KubeletReadOnlyPort > 0 {
+		return nil, k8sErr.New("cannot use both the read-only and secure port")
+	}
+	kubeletPort := config.KubeletReadOnlyPort
+	scheme := "http"
+	if kubeletPort <= 0 {
+		kubeletPort = config.KubeletSecurePort
+		scheme = "https"
+	}
+	if kubeletPort <= 0 {
+		kubeletPort = defaultSecureKubeletPort
+		scheme = "https"
+	}
+
+	kubeletURL := url.URL{
+		Scheme: scheme,
+		Host:   fmt.Sprintf("127.0.0.1:%d", kubeletPort),
```

Comment:
I think this need to be configurable.  Using localhost only works with host networking.

I'd prefer to not do that, and expose the nodeName to spire-agent via an environment variable, like this:

```          env:
            - name: MY_NODE_NAME
              valueFrom:
                fieldRef:
                  fieldPath: spec.nodeName
```

If spire-agent could read directly from that env var, that would work best, but otherwise a script could template the config at agent startup.

Reply:
That's a great point. I will fix it to take the envvar to use from a configurable `node_name_env` value (defaulting to `MY_NODE_NAME` since that is prevalent in the docs) and also support a `node_name` configurable for setting directly. If the node name is not in the env and `node_name` isn't set, I'll default to localhost.

Reply:
And I think I'll have `node_name` will trump `node_name_env`.